### ISSUE TITLE
Add ssh protocol

### ISSUE THREAD

**Issue by bbyars** [bbyars/mountebank#16]
Add ssh protocol, primarily for mock verification.  From Max Lincoln:

SSH would be a very powerful protocol to use.  A lot of the new infra-as-code cloud tools basically:
1. Use REST to call cloud providers to build servers/load balancers/etc
2. (Optionally) Send data from the cloud provider to a management tool, like Hosted Chef, using REST again.
3. Then SSH to the server(s) for configuration, running commands and/or using SSH-based file-transfer protocols.

1 & 2 are easily stubbed, but 3 is not.

What's awesome is that apparently [writing an rsync server in node](https://github.com/rvagg/node-libssh) is easy, and sftp seems a little more complicated, but scp and rsync just send simple commands to start a server, and gives clues about the file it's going to receive:
Sending command: scp -v -t asdf
Sending command: rsync --server -v . 22:asdf

In other words, SSH looks very recordable/mockable.

**Comment by daluu** [comment]
I'm not well versed in SSH and node, but would the security/cryptography part be easy to handle/mock? Unlike HTTP, SSH deals with login credentials and in other more secure cases, public/private keys. Or I guess that's just config steps as part of the setup for the keys and stuff.

**Comment by bbyars** [comment]
Yeah I suspect it could be config, but you're right that there's definitely
some tricky bits to work out.  I haven't looked into what it would take in
depth.
